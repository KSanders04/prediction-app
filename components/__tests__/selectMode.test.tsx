import { render, fireEvent } from '@testing-library/react-native';
import { router } from 'expo-router';
import SelectedMode from '../../app/selectMode';

// Mock expo-router
jest.mock('expo-router', () => ({
  router: {
    push: jest.fn(),
  },
}));

describe('SelectedMode Component', () => {
  beforeEach(() => {
    // Clear mock calls between tests
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { getByText } = render(<SelectedMode />);
    
    // Check if all buttons and title are present
    expect(getByText('Select Game Mode')).toBeTruthy();
    expect(getByText('Solo')).toBeTruthy();
    expect(getByText('Groups')).toBeTruthy();
    expect(getByText('Legend')).toBeTruthy();
  });

  it('navigates to home when Solo is pressed', () => {
    const { getByText } = render(<SelectedMode />);
    const soloButton = getByText('Solo');
    
    fireEvent.press(soloButton);
    expect(router.push).toHaveBeenCalledWith('/home');
  });

  it('navigates to groupMode when Groups is pressed', () => {
    const { getByText } = render(<SelectedMode />);
    const groupsButton = getByText('Groups');
    
    fireEvent.press(groupsButton);
    expect(router.push).toHaveBeenCalledWith('/groupMode');
  });

  it('navigates to empty route when Legend is pressed', () => {
    const { getByText } = render(<SelectedMode />);
    const legendButton = getByText('Legend');
    
    fireEvent.press(legendButton);
    expect(router.push).toHaveBeenCalledWith('/');
  });

  it('matches snapshot', () => {
    const tree = render(<SelectedMode />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});