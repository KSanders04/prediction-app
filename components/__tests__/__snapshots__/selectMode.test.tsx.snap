// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMode Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 28,
        "fontWeight": "700",
        "marginBottom": 40,
      }
    }
  >
    Select Game Mode
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3949AB",
        "borderRadius": 12,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Solo
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3949AB",
        "borderRadius": 12,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Groups
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3949AB",
        "borderRadius": 12,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Legend
    </Text>
  </View>
</View>
`;
